### ISSUE TITLE
"No multiple set in the same block" trigger when eth_estimiateGas call on go-ethereum 1.12.0

### ISSUE THREAD

**Issue by chun-ko** [iden3/contracts#172]
https://github.com/iden3/contracts/blob/860b3acb144515c45593053e8b05ae80afcefdf9/contracts/state/State.sol#L387

I deploy the iden3 protocol on go-ethereum 1.12.0, and execute the `transitState` in State.sol (via polygonid js-sdk), "No multiple set in the same block" is triggered.

This error does not trigger on polygon mumbai and go-ethereum version on 1.10.x

Then I come across the following fix in the go-ethereum 1.12.0 https://github.com/ethereum/go-ethereum/pull/24363/files, which changes the `block.number` from PendingBlockNumber to LatestBlockNumber.

I have modified the code from
```
prevStateInfo.createdAtBlock != block.number,
```
to
```
prevStateInfo.createdAtBlock != (block.number + 1),
```
And it working fine.

My understand the current IDEN3 protocol has been deployed on Polygon networks. Not sure if this has been deployed in Ethereum.

**Comment by OBrezhniev** [comment]
Hi @chun-ko! 

Thanks for reporting this issue!

Proposed code change
> ```
> prevStateInfo.createdAtBlock != (block.number + 1),
> ```
doesn't make much sense because it will never be equal.

Better remove this check in your contract for now.

We will evaluate security implications of removing it and do it if everything would look ok.

You are right that IDEN3 contracts are currently deployed on Polygon networks. And they are not deployed to Ethereum (we've tested on Ethereum testnets though and everything works).

**Comment by OBrezhniev** [comment]
Fixed in #176
